controllers/growth.js

```diff
@@ -3,45 +3,208 @@
-            }
+        if (query_string.from && moment(query_string.from).isValid()) {
+            from = moment(query_string.from).format('YYYY-MM-DD');
+        }
 
-            if (req.query.to && moment(req.query.to).isValid()) {
-                to =  moment(req.query.to).format('YYYY-MM-DD');
-            }
+        if (query_string.to && moment(query_string.to).isValid()) {
+            to =  moment(query_string.to).format('YYYY-MM-DD');
+        }
+
+        if (query_string.to && query_string.from) {
+            get_stat(null, []);
+        }
+
+        mysql.open(config.DB)
+            .query('SELECT channel_id, title, network, linked_date FROM mcn_channels '
+                   + 'WHERE channel_id = ?',
+                   [data.id],
+                   get_stat
+            )
+            .end();
+    }
+
+    function get_stat (err, result) {
+        if (err) {
+            return next(err);
+        }
+
+        if (result.length) {
+            from = moment(result[0].linked_date).format('YYYY-MM-DD');
+            to = moment(result[0].linked_date).add(2, 'month').format('YYYY-MM-DD');
+        }
+
+        mysql.open(config.DB)
+            .query(
+                'SELECT * FROM channel_stats WHERE channel_id = ? '
+                + 'AND insert_date BETWEEN ? AND ? ORDER BY insert_date',
+                [data.id, from, to],
+                send_response
+            )
+            .end();
+    }
+
+    function send_response (err, result) {
+        if (err) {
+            return next(err);
+        }
+
+        if (!result.length) {
+            return res.warn(404, 'No channel statistics');
+        }
+
+        res.send(result);
+    }
+
+    start();
+};
+
+
+
+exports.get_grouped_channel = function (req, res, next) {
+    var date = moment(),
+        start_month = date.subtract(3, 'month').format('YYYY-MM'),
+        end_month = date,
+        data = util.get_data([], ['month', 'network',
+                             'duration'], req.query),
+        channels = [],
+        cache_key,
+        cache = new memcached(config.memcached.nodes);
+
+    function start () {
+        var simple_key;
+
+        if (data.month) {
+            end_month = moment(data.month + '-01').format('YYYY-MM');
+        }
+
+        if (+data.duration) {
+            start_month = moment(end_month + '-01')
+                .subtract(+data.duration, 'month').format('YYYY-MM');
+        }
+
+        console.log('from ' + start_month + ' to ' + end_month);
+        console.log(data.network);
+
+        simple_key = data.network.split(',') + ':' + start_month + ':' + end_month;
+        cache_key = crypto.createHash('md5').update(simple_key).digest('hex');
 
-            mysql.open(config.DB)
-                .query(
-                    query, [data.id, from, to],
-                    send_response
+        // check cache first
+        cache.get(cache_key, retrieve_cache);
+    }
+
+    function retrieve_cache (err, cache_result) {
+
```

Comment:
be consistent with empty lines as first line of functions if you want empty lines